katsdpimager/loader_katdal.py

```diff
@@ -167,18 +175,24 @@ def __init__(self, filename, options, start_channel, stop_channel):
         # Find baselines represented by corr_products.
         baselines = _unique((a[:-1], b[:-1]) for a, b in self._file.corr_products)
         corr_product_permutation = []
-        missing_corr_products = []
         for a, b in baselines:
             for pol in pols:
                 corr_product = (a + pol[0], b + pol[1])
                 corr_product_permutation.append(corr_product_inverse.get(corr_product, None))
+        missing_corr_products_mask = np.zeros(len(corr_product_permutation), np.bool_)
```

Comment:
It's only `np.bool` that's deprecated. I have a mild preference for `np.bool_` since that's the actual type of the scalars in the array:
```python
>>> type(np.zeros(1, bool)[0])
numpy.bool_
```

Reply:
Fair enough. I was swayed by this sentence in the NumPy 1.20.0 release notes:

> To give a clear guideline for the vast majority of cases, for the types bool, object, str (and unicode) using the plain version is shorter and clear, and generally a good replacement.

Reply:
Yes, I saw that as well. If it soothes your OCD I'm happy to accept a PR, but I don't feel any need to do it myself.

Reply:
Nah :-)